frontend/src/pages/Profiles/Client/ClientArea.jsx

```diff
@@ -7,21 +7,66 @@ import {
     Text,
     Table,
     Button,
-    Heading
+    Heading,
+    toaster
 } from "evergreen-ui";
 import imgclient from "./images.png";
 import StarRatingComponent from "react-star-rating-component";
 import HeaderBar from "../../../components/HeaderBar";
+import { doGetSelfJobs, doGetJobDetail } from "../../../data/api";
 
 class ClientProfileArea extends React.Component {
     constructor() {
         super();
-        this.state = { selectedIndex: 0 };
+        this.state = {
+            selectedIndex: 0,
+            jobs: [],
+            freelancers: {}
+        };
+    }
+    componentDidMount() {
+        doGetSelfJobs()
+            .then(body => {
+                this.setState({
+                    jobs: body.jobs
+                });
+                let newState = {};
+                this.state.jobs.forEach(element => {
```

Comment:
This is better; this.setState({}) may not change the this.state instantly. 
```suggestion
                body.jobs.forEach(element => {
```